components/prize_pool/wikis/commons/starcraft_starcraft2/prize_pool_starcraft.lua

```diff
@@ -0,0 +1,357 @@
+local OPPONENT_TYPE_TO_MODE = {
+	solo = '1v1',
+	duo = '2v2',
+	trio = '3v3',
+	quad = '4v4',
+	team = 'team',
+}
+local PLACE_TO_KEY_PREFIX = {'winner', 'runnerup', 'third', 'fourth'}
+local SEMIFINALS_PREFIX = 'sf'
+local TBD = 'TBD'
+local SC2 = 'starcraft2'
+
+local _lpdb_stash = {}
+local _series
+local _tier
+local _tournament_extradata_cache = {{}, {}, {}, {}, ['3-4'] = {}}
+local _tournament_name
+
+-- Template entry point
+function CustomPrizePool.run(frame)
+	local args = Arguments.getArgs(frame)
+
+	-- use different opponent modules
+	args.opponentLibrary = 'Opponent/Starcraft'
+	args.opponentDisplayLibrary = 'OpponentDisplay/Starcraft'
+
+	-- set some default values
+	args.prizesummary = Logic.emptyOr(args.prizesummary, false)
+	args.exchangeinfo = Logic.emptyOr(args.exchangeinfo, false)
+	args.storelpdb = Logic.emptyOr(args.storelpdb, Namespace.isMain())
+	args.storesmw = Logic.emptyOr(args.storesmw, Namespace.isMain())
+	args.syncPlayers = Logic.emptyOr(args.syncPlayers, true)
+
+	-- overwrite some wiki vars for this PrizePool call
+	_tournament_name = args['tournament name']
+	_series = args.series
+	_tier = args.tier or Variables.varDefault('tournament_liquipediatier')
+
+	-- adjust import settings params
+	args.importLimit = tonumber(args.importLimit) or CustomPrizePool._defaultImportLimit()
+	args.allGroupsUseWdl = Logic.emptyOr(args.allGroupsUseWdl, true)
+	args.import = Logic.emptyOr(args.import, true)
+
+	-- fixed setting
+	args.resolveRedirect = true
+
+	local prizePool = PrizePool(args):create()
+
+	prizePool:setLpdbInjector(CustomLpdbInjector())
+	prizePool:setSmwInjector(CustomSmwInjector())
+
+	local buildPrizePool = prizePool:build()
+
+	local prizePoolIndex = tonumber(Variables.varDefault('prizepool_index')) or 0
+	-- set an additional wiki-var for legacy reasons so that combination with award prize pools still work
+	Variables.varDefine('prize pool table id', prizePoolIndex)
+	if prizePoolIndex == 1 and Logic.readBool(Logic.emptyOr(args.storeTournament, Namespace.isMain())) then
+		CustomPrizePool._storeIntoTournamentLpdb()
+	end
+
+	if Logic.readBool(args.storelpdb) then
+		pageVars:set('placementRecords.' .. prizePoolIndex, Json.stringify(_lpdb_stash))
+	end
+
+	return buildPrizePool
+end
+
+function CustomLpdbInjector:adjust(lpdbData, placement, opponent)
+	local seriesNumber = tonumber(Variables.varDefault('tournament_series_number'))
+	local lastStatuses = {
+		CustomPrizePool._getStatusFromScore(lpdbData.lastscore),
+		CustomPrizePool._getStatusFromScore(lpdbData.lastvsscore),
+	}
+	local extradata = {
+		featured = Variables.varDefault('featured') or 'false',
+		lastStatuses = lastStatuses,
+		placeRange = {placement.placeStart, placement.placeEnd},
+		playernumber = StarcraftOpponent.partySize(opponent.opponentData),
+		points = placement:getPrizeRewardForOpponent(opponent, PRIZE_TYPE_POINTS .. 1),
+		points2 = placement:getPrizeRewardForOpponent(opponent, PRIZE_TYPE_POINTS .. 2),
```

Comment:
atm no, we query them and also condition on them
longterm we want standardized points storage anyways, so i would rather switch then instead of now so not having to rewrite the modules using the data several times